src/main/java/io/cryostat/jfr/datasource/server/Datasource.java

```diff
@@ -154,7 +156,8 @@ void upload(RoutingContext context) {
 
         final StringBuilder responseBuilder = new StringBuilder();
 
-        uploadFiles(context.fileUploads(), responseBuilder);
+        boolean overwrite = Boolean.valueOf(extractQueryParam(context, "overwrite", "false"));
```

Comment:
https://docs.oracle.com/javase/7/docs/api/java/lang/Boolean.html#parseBoolean(java.lang.String)
https://docs.oracle.com/javase/7/docs/api/java/lang/Boolean.html#valueOf(java.lang.String)

Not too important, but `valueOf` returns boxed `Boolean` and `parseBoolean` returns a primitive `boolean`.

Reply:
I guess using primitive boolean saving some memory space?